Libplanet/Blockchain/BlockChain.cs

```diff
@@ -182,17 +187,18 @@ bool render
         /// <summary>
         /// An event which is invoked when <see cref="Tip"/> is changed.
         /// </summary>
-        public event EventHandler<TipChangedEventArgs> TipChanged;
-
-        /// <summary>
-        /// An event which is invoked when the chain is reorged.
-        /// </summary>
-        public event EventHandler<ReorgedEventArgs> Reorged;
+        private event EventHandler<(Block<T> OldTip, Block<T> NewTip)> TipChanged;
```

Comment:
> This pull request introduces a new interface named IRenderer<T> and replaces rendering methods of IAction and TipChanged/Reorged events on BlockChain<T> with it.

Just curiosity... Is this change make replaces `TipChanged` event from `BlockChain<T>` to `IRenderer<T>`?

Reply:
> Just curiosity... Is this change make replaces `TipChanged` event from `BlockChain<T>` to `IRenderer<T>`? 

Yes!:

> Besides rendering `IAction`s, `IRenderer<T>` also replaces `BlockChain<T>.TipChanged` event (introduced in #517 & #526) with `IRenderer<T>.RenderBlock()` method, and `BlockChain<T>.Reorged` event (introduced in #945) with `IRenderer<T>.RenderReorg()` method.  It means now all events regarding state changes on `BlockChain<T>` become unified into one interface `IRenderer<T>`.
> 
> (FYI I actually left `BlockChain<T>.TipChanged` event `private` for an internal use: cancelling undergoing mining when a new block is appended.  However, still it is invisible from public surface.)

Reply:
Ah, I understand. Move rendering role from `TipChanged` event to `IRenderer<T>.RenderBlock()` and `TipChanged` have only check changed tip blockchain itself. right?

Reply:
@riemannulus Right!